Skip tidy triagebot linkcheck if triagebot.toml doesn't exist

Distribution tarballs won't include triagebot.toml, which
causes tidy checks to fail with the following error:

   tidy error: triagebot.toml file not found
   some tidy checks failed

Drop the triagebot.toml check which is mostly to catch broken
path filters *within* `triagebot.toml` and not enforce
the existence of it.

Upstream-Status: Backport [https://github.com/rust-lang/rust/pull/142666/commits]

Signed-off-by: Yash Shinde <Yash.Shinde@windriver.com>
---
diff --git a/src/tools/tidy/src/triagebot.rs b/src/tools/tidy/src/triagebot.rs
--- a/src/tools/tidy/src/triagebot.rs
+++ b/src/tools/tidy/src/triagebot.rs
@@ -6,8 +6,11 @@
 
 pub fn check(path: &Path, bad: &mut bool) {
     let triagebot_path = path.join("triagebot.toml");
+
+    // This check is mostly to catch broken path filters *within* `triagebot.toml`, and not enforce
+    // the existence of `triagebot.toml` itself (which is more obvious), as distribution tarballs
+    // will not include non-essential bits like `triagebot.toml`.
     if !triagebot_path.exists() {
-        tidy_error!(bad, "triagebot.toml file not found");
         return;
     }

